src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketCredentials.java

```diff
@@ -48,14 +50,38 @@
 /**
  * Utility class for common code accessing credentials
  */
-class BitbucketCredentials {
-    private BitbucketCredentials() {
-        throw new IllegalAccessError("Utility class");
+class BitbucketCredentials implements Serializable {
+
+    private static final long serialVersionUID = 8439574104406702534L;
+    private final String serverUrl;
+    private final String contextName;
+    //private final SCMSourceOwner context;
+    private final String id;
+
+    public BitbucketCredentials(@CheckForNull String serverUrl,
+                                @CheckForNull String contextName,
+                                @CheckForNull String id) {
+        this.serverUrl = serverUrl;
+        this.contextName = contextName;
+        this.id = id;
+
+    }
+
+    public <T extends StandardCredentials> T credentials(@NonNull Class<T> type){
+        Jenkins instance = Jenkins.getInstanceOrNull();
+        if (instance == null) {
+            return null;
+        }
+        Item context = instance.getItemByFullName(contextName);
+        return lookupCredentials(serverUrl,context, id, type);
+    }
+
+    public StandardCredentials credentials(){
+        return credentials(StandardCredentials.class);
     }
 
-    @CheckForNull
     static <T extends StandardCredentials> T lookupCredentials(@CheckForNull String serverUrl,
-                                                               @CheckForNull SCMSourceOwner context,
+                                                               @CheckForNull Item context,
```

Comment:
I'm not sure this is a correct change, what was the rationale behind this - esp as `fillCredentials` is using the `SCMSourceOwner`?